### PR TITLE
Create crud requirement_type

### PR DESCRIPTION
**Pull request by jolivo07**

### CODE REVIEW

File: app/routes.go [hunk 1]
```diff
@@ -4,19 +4,28 @@ import (
...
 	root.Use(middleware.RequestID)
 	root.Use(middleware.Database)
 	root.Use(middleware.ParameterLogger)
 	root.Use(middleware.CSRF)
 
 	root.GET("/", home.Index)
+	root.GET("/requirement-type", requirement_type.List)
+	root.GET("/requirement-type/new", requirement_type.New)
+	root.POST("/requirement-type/create", requirement_type.Create)
+	root.GET("/Show/{requirement_type_id}", requirement_type.Show)
```
Comment (rdiazrovira) [inline app/routes.go]: Please make this route fully lowercase.

File: app/models/requirement_type.go [hunk 2]
```diff
@@ -6,10 +6,13 @@ import (
...
+// RequirementType model struct.
 type RequirementType struct {
-	ID           uuid.UUID `db:"id"`
-	DepartmentId uuid.UUID `db:"department_id"`
-	Name         string    `db:"name"`
-	CreatedAt    time.Time `db:"created_at"`
-	UpdatedAt    time.Time `db:"updated_at"`
+	ID           uuid.UUID `form:"id" db:"id"`
+	DepartmentID uuid.UUID `form:"department_id" db:"department_id"`
+	Name         string    `form:"name" db:"name" fako:"domain_name"`
```
Comment (rdiazrovira) [inline app/models/requirement_type.go]: What is "domain_name"?

File: app/actions/requeriment_type/requirement_type_test.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	}
+
+	as.NoError(as.DB.Create(&department))
+
+	requirementType := models.RequirementType{}
+	fako.Fill(&requirementType)
+	requirementType.DepartmentID = department.ID
+	as.NoError(as.DB.Create(&requirementType))
+
+	res := as.HTML("/requirement-type/").Get()
```
Comment (rdiazrovira) [inline app/actions/requeriment_type/requirement_type_test.go]: This route should be named "/requirement-types/". Plural :)

File: app/actions/requeriment_type/requirement_type_test.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+	requirementType.DepartmentID = department.ID
+	fako.Fill(&requirementType)
+
+	res := as.HTML("/requirement-type/create/").Post(requirementType)
+
+	as.Equal(res.Code, http.StatusSeeOther)
+
+	as.Equal("/requirement-type/", res.Location())
+
+	RequirementTypes := models.RequirementTypes{}
```
Comment (rdiazrovira) [inline app/actions/requeriment_type/requirement_type_test.go]: Please fix the name, the first character should be in lowercase.

File: app/actions/requeriment_type/requirement_type_test.go [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+	requirementTypeUpdate := &models.RequirementType{}
+	fako.Fill(requirementTypeUpdate)
+	requirementTypeUpdate.DepartmentID = department.ID
+
+	res := as.HTML("/update/" + requirementType.ID.String()).Put(requirementTypeUpdate)
+	as.Equal(res.Code, http.StatusSeeOther)
+	as.Equal("/requirement-type/", res.Location())
+
+}
```
Comment (rdiazrovira) [inline app/actions/requeriment_type/requirement_type_test.go]: Can you validate that the requirement type has been actually updated?

File: app/actions/requeriment_type/requirement_type_test.go [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+	fako.Fill(requirementType)
+	as.NoError(as.DB.Create(requirementType))
+
+	res := as.HTML("/delete/" + requirementType.ID.String()).Delete()
+	body := res.Body.String()
+	as.NotContains(body, requirementType.Name)
+	as.Equal(res.Code, http.StatusSeeOther)
+	as.Equal("/requirement-type/", res.Location())
+
+}
```
Comment (rdiazrovira) [inline app/actions/requeriment_type/requirement_type_test.go]: Please validate that the requirement type has been actually deleted.

**Review by rdiazrovira — CHANGES_REQUESTED** [review]
Take a look at those comments I left here.

**Review by rdiazrovira — APPROVED** [review]

**Review by mperezlamadrid — APPROVED** [review]

**Pull request merged**